File: app/src/main/java/com/esorokin/lantector/model/network/exception/UnhandledApiException.java
package com.esorokin.lantector.model.network.exception;

import com.esorokin.lantector.model.network.data.ApiStatusCode;

import lombok.Getter;

@Getter
public class UnhandledApiException extends ApiException {
	private String message;

	public UnhandledApiException(String message) {
		super(ApiStatusCode.UNDEFINED);
		this.message = message;
	}
}

File: app/src/main/java/com/esorokin/lantector/model/network/exception/UnhandledApiException.kt
package com.esorokin.lantector.model.network.exception

import com.esorokin.lantector.model.network.data.ApiStatusCode

class UnhandledApiException(message: String) : ApiException(ApiStatusCode.UNDEFINED) {
    constructor(exception: Exception) : this(exception.localizedMessage)
}
